web3_proxy/src/admin_queries.rs

```diff
@@ -0,0 +1,134 @@
+            StatusCode::BAD_REQUEST,
+            "Unable to get the user_tier_title key from the request".to_string(),
+            None,
+        ))?;
+
+    // Prepare output body
+    let mut response_body = HashMap::new();
+
+    // Establish connections
+    let db_conn = app.db_conn().context("query_admin_modify_user needs a db")?;
+    let db_replica = app
+        .db_replica()
+        .context("query_user_stats needs a db replica")?;
+    let mut redis_conn = app
+        .redis_conn()
+        .await
+        .context("query_admin_modify_user had a redis connection error")?
+        .context("query_admin_modify_user needs a redis")?;
+
+    // Try to get the user who is calling from redis (if existent) / else from the database
+    // TODO: Make a single query, where you retrieve the user, and directly from it the secondary user (otherwise we do two jumpy, which is unnecessary)
+    // get the user id first. if it is 0, we should use a cache on the app
+    let caller_id = get_user_id_from_params(&mut redis_conn, &db_conn, &db_replica, bearer, &params).await?;
+
+    // Check if the caller is an admin (i.e. if he is in an admin table)
+    let admin: admin::Model = admin::Entity::find()
+        .filter(admin::Column::UserId.eq(caller_id))
+        .one(&db_conn)
+        .await?
+        .ok_or(AccessDenied.into())?;
+
+    // If we are here, that means an admin was found, and we can safely proceed
+
+    // Fetch the admin, and the user
+    let user: user::Model = user::Entity::find()
+        .filter(user::Column::Address.eq(user_address))
+        .one(&db_conn)
+        .await?
+        .ok_or(FrontendErrorResponse::StatusCode(
+            StatusCode::BAD_REQUEST,
+            "No user with this id found".to_string(),
+            None,
+        ))?;
+    // Return early if the target user_tier_id is the same as the original user_tier_id
+    response_body.insert(
+        "user_tier_title",
+        serde_json::Value::Number(user.user_tier_id.into()),
+    );
+
+    // Now we can modify the user's tier
+    let new_user_tier: user_tier::Model = !user_tier::Entity::find()
+        .filter(user_tier::Column::Title.eq(user_tier_title.clone()))
+        .one(&db_conn)
+        .await?
+        .ok_or(|| FrontendErrorResponse::StatusCode(
+            StatusCode::BAD_REQUEST,
+            "User Tier name was not found".to_string(),
+            None,
+        ))?;
+
+    if user.user_tier_id == new_user_tier.id {
+        info!("user already has that tier");
+    } else {
+        let mut user = user.into_active_model();
+
+        user.user_tier_id = sea_orm::Set(new_user_tier.id);
+
+        user.save(&db_conn).await?;
+
+        info!("user's tier changed");
+    }
+
+    // Finally, remove the user from redis
+    // TODO: Also remove the user from the redis
+    // redis_conn.zrem();
+    // redis_conn.get::<_, u64>(&user.) // TODO: Where do i find the bearer token ...
```

Comment:
I think you'll need to do a query on the [login](https://github.com/llamanodes/web3-proxy/blob/main/entities/src/login.rs) table. That should let you get all bearer tokens for the user.

I think you'll want to delete them from redis and from the database.